lib/jazzy/doc_builder.rb

```diff
@@ -156,6 +173,21 @@ def self.decl_for_token(token)
       end
     end
 
+    def self.puts_lint(undocumented)
+      tokens_by_file = undocumented.group_by do |d|
+        if d['key.filepath']
+          Pathname.new(d['key.filepath']).basename.to_s
+        else
+          d['key.modulename'] || ''
+        end
+      end
+      tokens_by_file.each_key do |file|
```

Comment:
I think that this structure is a hash of arrays, so the logic is iterating over the hash keys first and then the arrays. Is there a better way to structure this in ruby?